File: mobile/lib/services/api_service.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

class ApiService {
  static const baseUrl = "http://10.0.2.2:8000";

  static Future<List<String>> textToSign(String text) async {
    final response = await http.post(
      Uri.parse("$baseUrl/text-to-sign"),
      body: {"text": text},
    );

    final data = jsonDecode(response.body);
    return List<String>.from(data["sequence"]);
  }
}

File: mobile/src/services/ApiService.ts
import { BASE_URL } from "../config";
import { WordModel, LandmarkPoint } from "../models/SignModel";

export async function textToSign(text: string): Promise<WordModel[]> {
  const response = await fetch(
    `${BASE_URL}/text-to-sign?text=${encodeURIComponent(text)}`,
    { method: "POST" }
  );
  const data = await response.json();
  return data.words as WordModel[];
}

export async function fetchSignLandmarks(letter: string): Promise<LandmarkPoint[]> {
  const response = await fetch(`${BASE_URL}/signs/${letter.toLowerCase()}/landmarks`);
  if (!response.ok) throw new Error(`No landmarks for '${letter}'`);
  const data = await response.json();
  return data.landmarks as LandmarkPoint[];
}
